### ISSUE TITLE
String allow null conversion

### ISSUE THREAD

**Issue by martinmacko47** [Twipped/joi-to-swagger#24]
Is conversion `joi.string().allow(null)` to`"type": ["string", "null"]` a valid value for swagger type? Because swagger UI (at least [swagger-ui-express](https://www.npmjs.com/package/swagger-ui-express)) does not seem to understand it. Instead, it writes `"Unknown Type: string,null"`.

Although didn't find any mention of multiple types in [swagger 2.0 spec](https://swagger.io/docs/specification/2-0/describing-request-body/), all used examples show just a single type value.

And in [swagger 3.0 spec](https://swagger.io/docs/specification/data-models/data-types/) it's explicitely said that:

> `type` takes a single value. `type` as a list is not valid in OpenAPI (even though it is valid in JSON Schema):

**Comment by Twipped** [comment]
You're right, it's not valid swagger by the spec. This was an extension to swagger that we used at Shutterstock because otherwise there was no way to define a property that could be null. The libraries we used back then supported it (curiously, I thought Swagger UI was one of them).

There is also no null type. Swagger 3.0 added the `nullable: true` attribute. I don't have the time right now to change this lib to use it, but I'd welcome a pull request that switched to that behavior.

**Comment by martinmacko47** [comment]
You mean a PR to add `nullable: true` instead of `type: ["string", "null"]`, or to remove `null` from `type: ["string", "null"]`.

To which swagger verion `joi-to-swagger` targets? To 2.0 or to 3.0? So we won't start generating invalid schema again as swagger 2.0 does not support `nullable: true`.

**Comment by Twipped** [comment]
The former.

IMO Swagger 2 is a fundamentally flawed spec, I see no reason to continue to support it now that 3 is out. This would probably a change meriting a major version bump, however.

**Comment by Twipped** [comment]
Fixed in 2.0 release

**Comment by deskoh** [comment]
See https://github.com/ajv-validator/ajv/issues/1471. Enum doesn't play well with `nullable`.

This schema

```js
const schema = joi
  .object({
    field1: joi.valid("a", "b", null)
  })
  .unknown();
```

below generates:

```json
{
  "type": "object",
  "properties": {
    "field1": {
      "enum": ["a", "b"],
      "nullable": true
    }
  }
} 
```

The correct schema should be
```json
{
  "type": "object",
  "properties": {
    "field1": {
      "enum": ["a", "b", null]
      "nullable": true
    }
  }
} 
```

> `nullable` flag and `null` are required

Codesandbox link https://codesandbox.io/s/json-schema-enum-nullable-zsrlbj?file=/src/index.test.js
